3.4/s2i/bin/run

```diff
@@ -35,14 +35,34 @@ function get_default_web_concurrency() {
   echo $default
 }
 
+function maybe_run_in_init_wrapper() {
+  if [[ -z "$ENABLE_INIT_WRAPPER" ]]; then
+    exec $@
+  else
+    local wrapper=`mktemp -p .`
+    chmod u+x $wrapper
+    cat <<EOF > $wrapper
+trap 'kill -TERM \$PID' TERM INT
+$@
```

Comment:
Missing double quotes too.

This is rather a preference, but some precisely shaped init wrapper could live directly in e.g. shareable '/bin/s2i-init-wrapper' (could live in s2i-core).  Basically, at least I would prefer not to have temporary file for that; alternative could be `/bin/sh -c 'kill ... ; "$@" ; ...' -- "$@"`

Reply:
... and the `$ENABLE_INIT_WRAPPER` (or something else) could have the wrapper inside.